kf_lib_data_ingest/common/pandas_utils.py

```diff
@@ -244,8 +252,19 @@ def resolve_duplicates(df, suffixes):
         return df
 
     merged = pandas.merge(left, right, **kwargs)
+    reduced = resolve_duplicates(merged, kwargs.pop('suffixes', ('_x', '_y')))
+
+    msg = (
+        f'*** {kwargs.get("how", "inner").title()} merge {left_name} with '
+        f'{right_name}***\n'
+        f'-- {left_name} DataFrame Uniques --\n{left.nunique()}\n'
+        f'-- {right_name} DataFrame Uniques --\n{right.nunique()}\n'
+        f'-- Merged DataFrame Uniques --\n{reduced.nunique()}'
```

Comment:
What is the significance of comparing these values?

Reply:
When I was developing my transform module, I was printing these out after every merge. They were useful sanity checks for me to ensure I wasn't losing entities between merges. 

This brings me to another thought -- I developed my entire transform module in a Jupyter Notebook bc its so visual and useful to see the output df after every merge. I think it would be super neat if we accepted both .ipynb and .py files for transform modules. Personally, I feel like 90% of the time I will dev my transform module in a notebook anyway. And then I would copy it over to my_package/transform_module.py since that's all we accept right now. Also, since our ingest packages will be up on GitHub, if we allowed transform function notebooks, you could easily see the transformation code along with the data in a visually pleasing way. Thoughts?

Reply:
Committing notebook files is dangerous because of the risk of leaving data in them. I guess if you can figure out how to set it up so that they can be used but never committed to the repository then it would be ok.

Reply:
> I was printing these out after every merge. They were useful sanity checks for me to ensure I wasn't losing entities between merges.

But this doesn't show that. For that you'd need to stack left and right vertically first and compare nuniques for that version of them together vs the (hopefully) properly merged version of both of them together.

Reply:
> Committing notebook files is dangerous because of the risk of leaving data in them. I guess if you can figure out how to set it up so that they can never be committed to the repository then it would be ok.

Since this will be part of ingest package and kf-ingest-packages repository is private, I think it should be fine.

Reply:
> But this doesn't show that. For that you'd need to stack left and right vertically first and compare nuniques for that version of them together vs the merged version of them together. 

Yea you are correcto 👍